engine/apps/slack/alert_group_slack_service.py

```diff
@@ -0,0 +1,136 @@
+import logging
+
+from django.apps import apps
+
+from apps.slack.constants import SLACK_RATE_LIMIT_DELAY
+from apps.slack.slack_client import SlackClientWithErrorHandling
+from apps.slack.slack_client.exceptions import (
+    SlackAPIChannelArchivedException,
+    SlackAPIException,
+    SlackAPIRateLimitException,
+    SlackAPITokenException,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AlertGroupSlackService:
+    def __init__(self, slack_team_identity, slack_client=None) -> None:
+        self.slack_team_identuty = slack_team_identity
+        if slack_client is not None:
+            self.slack_client = slack_client
+        else:
+            self._slack_client = SlackClientWithErrorHandling(slack_team_identity.bot_access_token)
```

Comment:
do we need two slack client related instance attributes?
```suggestion
        self.slack_client = slack_client if slack_client else SlackClientWithErrorHandling(slack_team_identity.bot_access_token)
```

Reply:
Good catch, thanks